src/StakeRegistry.sol

```diff
@@ -203,15 +186,91 @@ contract StakeRegistry is VoteWeigherBase, StakeRegistryStorage {
+        _addStrategyParams(quorumNumber, strategyParams);
+        _setMinimumStakeForQuorum(quorumNumber, minimumStake);
+
+        _totalStakeHistory[quorumNumber].push(OperatorStakeUpdate({
+            updateBlockNumber: uint32(block.number),
+            nextUpdateBlockNumber: 0,
+            stake: 0
+        }));
+    }
 
-    /// @notice Adjusts the `minimumStakeFirstQuorum` -- i.e. the node stake (weight) requirement for inclusion in the 1st quorum.
-    function setMinimumStakeForQuorum(uint8 quorumNumber, uint96 minimumStake) external onlyServiceManagerOwner {
+    function setMinimumStakeForQuorum(
+        uint8 quorumNumber, 
+        uint96 minimumStake
+    ) public virtual onlyServiceManagerOwner quorumExists(quorumNumber) {
         _setMinimumStakeForQuorum(quorumNumber, minimumStake);
     }
 
+    /** 
+     * @notice Adds strategies and weights to the quorum
+     * @dev Checks to make sure that the *same* strategy cannot be added multiple times (checks against both against existing and new strategies).
+     * @dev This function has no check to make sure that the strategies for a single quorum have the same underlying asset. This is a concious choice,
+     * since a middleware may want, e.g., a stablecoin quorum that accepts USDC, USDT, DAI, etc. as underlying assets and trades them as "equivalent".
+     */
+    function addStrategies(
+        uint8 quorumNumber, 
+        StrategyAndWeightingMultiplier[] memory strategyParams
+    ) public virtual onlyServiceManagerOwner quorumExists(quorumNumber) {
+        _addStrategyParams(quorumNumber, strategyParams);
+    }
+
+    /**
+     * @notice Remove strategies and their associated weights from the quorum's considered strategies
+     * @dev higher indices should be *first* in the list of @param indicesToRemove, since otherwise
+     * the removal of lower index entries will cause a shift in the indices of the other strategies to remove
+     */
+    function removeStrategies(
+        uint8 quorumNumber,
+        uint256[] memory indicesToRemove
+    ) public virtual onlyServiceManagerOwner quorumExists(quorumNumber) {
+        uint256 toRemoveLength = indicesToRemove.length;
+        require(toRemoveLength > 0, "StakeRegistry.removeStrategyParams: no indices to remove provided");
+
+        StrategyAndWeightingMultiplier[] storage strategyParams = strategiesConsideredAndMultipliers[quorumNumber];
+
+        for (uint256 i = 0; i < toRemoveLength; i++) {
+            emit StrategyRemovedFromQuorum(quorumNumber, strategyParams[indicesToRemove[i]].strategy);
+            emit StrategyMultiplierUpdated(quorumNumber, strategyParams[indicesToRemove[i]].strategy, 0);
+
+            // Replace index to remove with the last item in the list, then pop the last item
+            strategyParams[indicesToRemove[i]] = strategyParams[strategyParams.length - 1];
+            strategyParams.pop();
+        }
+    }
```

Comment:
@wadealexc thoughts on the safety of this function?
it previously had a redundant parameter which it checked against, to ensure that the desired strategies were removed

Reply:
What redundant parameter? Here's the master branch: https://github.com/Layr-Labs/eigenlayer-middleware/blob/master/src/VoteWeigherBase.sol#L62

Reply:
I havent given any thought to the safety of the method... I think that'll come when we start working on integration tests. Essentially we should be testing that stake updates occur as expected even if we interweave strategy modifications in between updates.

Reply:
sorry, wasn't related to this PR, probably should have asked elsewhere, but here's a link to the previous behavior I was thinking of https://github.com/Layr-Labs/eigenlayer-contracts/blob/bd211f12fe8ed84f691ca44db3f812412defcb1e/src/contracts/middleware/VoteWeigherBase.sol#L134-L138

this could be described as an "idiot-check", but I do have a feeling it'd be easy to call this function "incorrectly" (or at least, in a way that doesn't have the desired outcome) with the way it works now